### PR TITLE
MODULES-1581: Gentoo compatibility

### PR DESCRIPTION
**Pull request by derdanne**
add Gentoo compatibility:
- adds params 
- adds rspec tests
- adds modifications in modules
- adds modification of Readme

### CODE REVIEW

**Comment by mhaskel** [comment]
@derdanne ping?

**Comment by derdanne** [comment]
Hi, sorry for the delay - i try to work on it in the next week.

**Comment by derdanne** [comment]
Hey,

i finished the requested tasks:

- rebased
- squashed my commits

I had to add a little workaround in Rakefile to not hit the issue to not ignore ignore_paths (https://github.com/rodjek/puppet-lint/issues/355) that caused a failure in lint testing.

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 ---
 language: ruby
 bundler_args: --without system_tests
+before_script:
+  - 'git clone git://github.com/gentoo/puppet-portage.git spec/fixtures/modules/portage'
+  - 'git clone git://github.com/puppetlabs/puppetlabs-concat.git spec/fixtures/modules/concat'
```
Comment (mhaskel) [inline .travis.yml]: Can't these just get added to .fixtures.yml?

File: Rakefile [hunk 2]
```diff
@@ -8,4 +8,12 @@ PuppetLint.configuration.send('disable_class_inherits_from_params_class')
...
-PuppetLint.configuration.ignore_paths = ["spec/**/*.pp", "pkg/**/*.pp"]
+
+exclude_paths = [
+"pkg/**/*",
+"spec/**/*",
+]
+
+Rake::Task[:lint].clear
+PuppetLint.configuration.ignore_paths = exclude_paths
+
```
Comment (mhaskel) [inline Rakefile]: Does this block need to go at the top of the file to make sure the other configurations aren't cleared?

**Comment by mhaskel** [comment]
Hi @derdanne, thanks for the update! I just have two questions/comments (inline), but otherwise looks good!

**Comment by derdanne** [comment]
Hi @mhaskel, you are totally right! I just fixed it.

**Comment by mhaskel** [comment]
Hi @derdanne,

With local testing the Rakefile updates actually breaks linting all together. Can you remove that change? We haven't encountered that issue on travis-ci as far as I know, so things should pass without it. Thanks!

**Comment by derdanne** [comment]
Hi @mhaskel, 

i reverted the changes in rakefile and travis ci doesn't break. 

It is really weird, testing breaked before without these changes on my local computer and in travis ci.

File: Rakefile [hunk 3]
```diff
@@ -1,7 +1,7 @@
 require 'puppetlabs_spec_helper/rake_tasks'
 require 'puppet-lint/tasks/puppet-lint'
 
-PuppetLint.configuration.fail_on_warnings = true
+PuppetLint.configuration.fail_on_warnings
```
Comment (mhaskel) [inline Rakefile]: Can you remove this change too?

**Comment by mhaskel** [comment]
@derdanne thanks for working through all the issues with this with me! Looks like there's just one more change then we should be able to get this merged.

**Comment by derdanne** [comment]
i reverted the change in Rakefile, now we got a merge conflict

**Comment by igalic** [comment]
thank you @derdanne!

**Pull request merged**